pkg/devfile/parser/parse.go

```diff
@@ -599,3 +605,78 @@ func setEndpoints(endpoints []v1.Endpoint) {
 		endpoints[i].Secure = &val
 	}
 }
+
+//parseKubeComponentFromURI iterate through all kubernetes components, and parse from uri and update the content to inlined field in devfileObj
+func parseKubeComponentFromURI(devObj DevfileObj) error {
+	getKubeCompOptions := common.DevfileOptions{
+		ComponentOptions: common.ComponentOptions{
+			ComponentType: v1.KubernetesComponentType,
+		},
+	}
+	kubeComponents, err := devObj.Data.GetComponents(getKubeCompOptions)
+	if err != nil {
+		return err
+	}
+	for _, kubeComp := range kubeComponents {
+		if kubeComp.Kubernetes.Uri != "" {
+			err := convertKubeCompUriToInlined(&kubeComp, devObj.Ctx)
```

Comment:
Is there a way to determine if the inlined content contains substitution variables?  If it doesn't then we don't need to process it

Reply:
it will still need to read in the content, and call `ValidateAndReplaceGlobalVariable` to look for global variable and validation. that will be too much extra work compare with convert all to inlined and only call `ValidateAndReplaceGlobalVariable` after parsing